src/transformers/file_utils.py

```diff
@@ -52,195 +55,88 @@
 ENV_VARS_TRUE_VALUES = {"1", "ON", "YES"}
 ENV_VARS_TRUE_AND_AUTO_VALUES = ENV_VARS_TRUE_VALUES.union({"AUTO"})
 
-try:
-    USE_TF = os.environ.get("USE_TF", "AUTO").upper()
-    USE_TORCH = os.environ.get("USE_TORCH", "AUTO").upper()
-    if USE_TORCH in ENV_VARS_TRUE_AND_AUTO_VALUES and USE_TF not in ENV_VARS_TRUE_VALUES:
-        import torch
-
-        _torch_available = True  # pylint: disable=invalid-name
-        logger.info("PyTorch version {} available.".format(torch.__version__))
-    else:
-        logger.info("Disabling PyTorch because USE_TF is set")
-        _torch_available = False
-except ImportError:
-    _torch_available = False  # pylint: disable=invalid-name
-
-try:
-    USE_TF = os.environ.get("USE_TF", "AUTO").upper()
-    USE_TORCH = os.environ.get("USE_TORCH", "AUTO").upper()
-
-    if USE_TF in ENV_VARS_TRUE_AND_AUTO_VALUES and USE_TORCH not in ENV_VARS_TRUE_VALUES:
-        import tensorflow as tf
-
-        assert hasattr(tf, "__version__") and int(tf.__version__[0]) >= 2
-        _tf_available = True  # pylint: disable=invalid-name
-        logger.info("TensorFlow version {} available.".format(tf.__version__))
-    else:
-        logger.info("Disabling Tensorflow because USE_TORCH is set")
-        _tf_available = False
-except (ImportError, AssertionError):
-    _tf_available = False  # pylint: disable=invalid-name
-
-
-try:
-    USE_JAX = os.environ.get("USE_FLAX", "AUTO").upper()
-
-    if USE_JAX in ENV_VARS_TRUE_AND_AUTO_VALUES:
-        import flax
-        import jax
-
-        logger.info("JAX version {}, Flax: available".format(jax.__version__))
-        logger.info("Flax available: {}".format(flax))
-        _flax_available = True
-    else:
-        _flax_available = False
-except ImportError:
-    _flax_available = False  # pylint: disable=invalid-name
-
-
-try:
-    import datasets  # noqa: F401
-
-    # Check we're not importing a "datasets" directory somewhere
-    _datasets_available = hasattr(datasets, "__version__") and hasattr(datasets, "load_dataset")
-    if _datasets_available:
-        logger.debug(f"Successfully imported datasets version {datasets.__version__}")
-    else:
-        logger.debug("Imported a datasets object but this doesn't seem to be the 🤗 datasets library.")
-
-except ImportError:
-    _datasets_available = False
-
-try:
-    from torch.hub import _get_torch_home
```

Comment:
This is the only breaking change in the PR. As we don't import `torch` at init, we can't do this anymore. I don't think it matters however, since this was just used for the old location of the cache directory, which is not used anymore (only copied if available).